pycyphal/transport/udp/_ip/_socket_factory.py

```diff
@@ -41,20 +39,21 @@ class SocketFactory(abc.ABC):
     """
 
     @staticmethod
-    def new(local_ip_address: IPAddress) -> SocketFactory:
+    def new(
+        local_ip_addr: typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address],
+        ipv6_addr: bool = False,
```

Comment:
```suggestion
```

This is not needed because `ipv6_addr = isinstance(local_ip_addr, ipaddress.IPv6Address)`